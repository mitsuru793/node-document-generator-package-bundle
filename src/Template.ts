import Mustache = require('mustache')
import { readSelfPackageJson, readTemplate } from './util'
import Package from './Package'

export default class Template {
  public static async readme(purpose: string): Promise<string> {
    const selfFull = readSelfPackageJson()
    const pkg = Package.parsePackageJson(selfFull)
    const dependencies = (await Package.fetchDependencies(selfFull)).sort()

    const template = readTemplate('README.md')
    const partials = {
      packageLinksLine: readTemplate('packageLinksLine.md')
    }
    const vars = {
      packageName: pkg.name,
      purpose,
      link: pkg.link,
      dev: true,
      dependencies
    }
    return Mustache.render(template, vars, partials).trim()
  }
}
