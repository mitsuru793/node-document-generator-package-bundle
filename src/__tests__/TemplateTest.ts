import Template from "../Template"

describe('Template', (): void => {
  it('can render readme', async () => {
    const text = await Template.readme('to play music')
    expect(text).toMatchSnapshot()
  })
})
