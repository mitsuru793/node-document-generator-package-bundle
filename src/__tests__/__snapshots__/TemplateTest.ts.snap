// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template can render readme 1`] = `
"# @mitsuru793/document-generator-package-bundle

[Npm](https://www.npmjs.com/package/@mitsuru793/document-generator-package-bundle) /
[Github](https://github.com/mitsuru793/node-document-generator-package-bundle) /
[Homepage](https://github.com/mitsuru793/node-document-generator-package-bundle)

This is a bundle of npm packages to play music.

You don't have look for npm packages to do it.

# Installation

Just install only one package!

\`\`\`
$ yarn add -D @mitsuru793/document-generator-package-bundle
\`\`\`

# Package List

You can use the following the packages without \`yarn add <the following package>\`.

### commander: 2.20.0

[Npm](https://www.npmjs.com/package/commander) /
[Github](https://github.com/tj/commander) /
[Homepage](https://github.com/tj/commander.js#readme)

the complete solution for node.js command-line programs

### dot-prop: 5.0.1

[Npm](https://www.npmjs.com/package/dot-prop) /
[Github](https://github.com/sindresorhus/dot-prop) /
[Homepage](https://github.com/sindresorhus/dot-prop#readme)

Get, set, or delete a property from a nested object using a dot path

### fs-extra: 8.0.1

[Npm](https://www.npmjs.com/package/fs-extra) /
[Github](https://github.com/jprichardson/node-fs-extra) /
[Homepage](https://github.com/jprichardson/node-fs-extra)

fs-extra contains methods that aren&#39;t included in the vanilla Node.js fs package. Such as mkdir -p, cp -r, and rm -rf.

### mustache: 3.0.1

[Npm](https://www.npmjs.com/package/mustache) /
[Github](https://github.com/janl/mustache) /
[Homepage](https://github.com/janl/mustache.js)

Logic-less {{mustache}} templates with JavaScript"
`;
